src/main/java/org/jboss/xavier/integrations/route/MainRouteBuilder.java

```diff
@@ -209,6 +222,15 @@ public void configure() {
                 .process(httpError403());
     }
 
+    private Exchange calculateICSAggregated(Exchange old, Exchange neu) {
+            UploadFormInputDataModel newObj = (UploadFormInputDataModel) neu.getIn().getBody(List.class).get(0);
+            UploadFormInputDataModel oldObj = (old != null) ? old.getIn().getHeader("uploadformdata", UploadFormInputDataModel.class) : null;
+            newObj.setTotalDiskSpace(newObj.getTotalDiskSpace() + ((old != null) ? oldObj.getTotalDiskSpace() : 0));
```

Comment:
Should this check be  `oldObj != null`?
```suggestion
            newObj.setTotalDiskSpace(newObj.getTotalDiskSpace() + ((oldObj != null) ? oldObj.getTotalDiskSpace() : 0));
```

Reply:
+1